doc/source/guide/in_depth/functional_guide.rst

```diff
@@ -0,0 +1,257 @@
+|                    |                 | - Retains all properties, if ``a`` is positive. |
++--------------------+-----------------+-------------------------------------------------+
+| ``(f * a)(x)``     | ``f(a * x)``    | `FunctionalRightScalarMult`                     |
+|                    |                 | - Retains all properties                        |
++--------------------+-----------------+-------------------------------------------------+
+| ``(v * f)(x)``     | ``v * f(x)``    | `FunctionalLeftVectorMult`                      |
+|                    |                 | - This is not a functional, but an operator     |
++--------------------+-----------------+-------------------------------------------------+
+| ``(f * v)(x)``     | ``f(v * x)``    | `FunctionalRightVectorMult`                     |
+|                    |                 | - Retains gradient and convex conjugate.        |
++--------------------+-----------------+-------------------------------------------------+
+| ``f.translate(y)`` | ``f(. - y)``    | `TranslatedFunctional`                          |
+|                    |                 | - Retains all properties.                       |
++--------------------+-----------------+-------------------------------------------------+
+
+
+Code example
+============
+This section contains an example of an implementation of a functional, namely the functional :math:`\|x\|_2^2 + \langle x, y \rangle`.
+Another example can be found `functional_basic_example.py`, and more implementations of other functionals can be found in `default_functionals.py`. ::
+
+    # Here we define the functional
+    class MyFunctional(odl.solvers.Functional):
+        """This is my functional: ||x||_2^2 + <x, y>."""
+
+        def __init__(self, domain, y):
+            # This comand calls the init of Functional and sets a number of
+            # parameters associated with a functional. All but domain have default
+            # values if not set.
+            super().__init__(domain=domain, linear=False, convex=True,
+                             concave=False, smooth=True, grad_lipschitz=2)
+
+            # We need to check that y is in the domain. Then we store the value of
+            # y for future use.
+            if y not in domain:
+                raise TypeError('y is not in the domain!')
+            self._y = y
+
+        # Property that returns the linear term.
+        @property
+        def y(self):
+            return self._y
+
+        # Defining the _call function
+        def _call(self, x):
+            return x.norm()**2 + x.inner(self.y)
+
+        # Next we define the gradient. Note that this is a property.
+        @property
+        def gradient(self):
+
+            # The class corresponding to the gradient operator.
+            class MyGradientOperator(odl.Operator):
+                """Class that implements the gradient operator of the functional
+                ``||x||_2^2 + <x,y>``.
```

Comment:
Should be one line here, too. If it doesn't fit, split the string into multiple lines.